### ISSUE TITLE
[files] Updating search index/database does not work without restarting

### ISSUE THREAD

**Issue by henrikgit** [albertlauncher/plugins#96]
#### Description
The search index/database isn't updated in a timely fashion, at least not by what the `scan interval` setting suggest. Basically, I'm confused and not sure about how "fast" the files' index, which are defined in the field above, are updated.

Or, which is absolutely a possibility, is it an unreasonable expectation to define my home as a path?

#### Expected behavior
I expect that after a few minutes, maybe 20 minutes at maximum, deleted files no longer get shown as a result in albert or new files get shown as a result.

#### Steps to reproduce
Not sure.

But here are my settings:

![2023-02-22-1438 albert-files plugin-settings](https://user-images.githubusercontent.com/6051516/220637285-a13bd6e3-4d78-484b-91f9-15021c35c5ad.png)

#### Source
AUR

#### Debug output

<details>

```
$ albert -d
Albert version: 0.20.8
Build date: Feb 21 2023 08:38:46
Qt version: 6.4.2
Build ABI: x86_64-little_endian-lp64
Arch (build/current): x86_64/x86_64
Kernel (type/version): linux/6.1.12-arch1-1
OS: Arch Linux
OS (type/version): arch/unknown
$QT_QPA_PLATFORMTHEME: 
Platform name: xcb
Font: Liberation Sans,11,-1,5,400,0,0,0,0,0,0,0,0,0,0,1
Binary location: /usr/bin/albert
$PWD: /home/john
$SHELL: /bin/bash
$LANG: de_DE.UTF-8
Language: German
Locale: de_DE
$XDG_SESSION_TYPE: x11
$XDG_CURRENT_DESKTOP: XFCE
$DESKTOP_SESSION: xfce
$XDG_SESSION_DESKTOP: xfce
Icon theme: gnome
14:40:49 [crit:albert] Application has not been terminated graciously
14:40:49 [debg:albert] Checking for a running instance…
14:40:49 [info:albert] There is another instance of albert running.
```

</details>

**Comment by ManuelSchneid3r** [comment]
I put a folder blabla in my homepath. set interval to one minute. a minute later the folder was available. deleted it. a minute later the folder is gone. interval seems to work.

<img width="283" alt="Bildschirm­foto 2023-02-23 um 13 44 33" src="https://user-images.githubusercontent.com/3252315/220909466-9ac65778-e528-4e5d-8630-4997c3054c32.png">

<img width="526" alt="Bildschirm­foto 2023-02-23 um 13 45 05" src="https://user-images.githubusercontent.com/3252315/220909669-b32f7a0d-0b22-4025-8f0a-f5b325c9ae54.png">

Either I missed the point or i cant reproduce the issue.

**Comment by henrikgit** [comment]
Thank you for your own test.

Well. Maybe I have to decrease the value for the max  depth. I don't know.

**Comment by ManuelSchneid3r** [comment]
Can you please provide a way to reproduce your issue?

**Comment by ManuelSchneid3r** [comment]
@henrikgit ?

**Comment by henrikgit** [comment]
Sorry.

Ok, just created a new folder `~/Videos/TestForAlbertManuel`. What should I do now?

**Comment by henrikgit** [comment]
In case the answer were "search for it after 1-2 minutes": Well, 2 minutes have passed and neither the full or partial name of the new folder gets found.

**Comment by henrikgit** [comment]
> As per the example in the post above i cant reproduce the issue. Can you please describe what you did to reveal the problem?

Ok now I did the following:

1. Closed albert
2. Restarted albert in the terminal with albert -d
3. Created `~/Videos/AnotherTestAlbert/`
4. Waited 2 minutes, maybe 2.5 minutes.
5. Hit the albert hotkey and started entering "AnotherTestAlbert".
6. The folder was not found.

> 
> the files should dissapear exactly after this interval is exceeded. if you run by `albert -d` from terminal the app prints an info when the indexing finished

and 

> does the output print that the folder has been indexed?

I did not see that specific output.



BUT: I just closed albert and restarted it. Now the new folder "AnotherTestAlbert" gets found instantly. Hm.

**Comment by ManuelSchneid3r** [comment]
> <img alt="Bildschirm­foto 2023-02-23 um 13 44 33" width="283" src="https://user-images.githubusercontent.com/3252315/220909466-9ac65778-e528-4e5d-8630-4997c3054c32.png">

nothing like this every minute? is your scan interval really set to 1 minut?

**Comment by henrikgit** [comment]
It really is. But yeah, there was that output. And here it is again.

![2023-03-10-1452 albert debug index](https://user-images.githubusercontent.com/6051516/224333463-888cb95a-fc40-4c49-a687-11909a694faf.png)

However right after I saw "Indexing home/john", I hit the hotkey, entered "LalaTes..." and it does not get found. I suspect it will get found after I restart albert.

**Comment by ManuelSchneid3r** [comment]
"LalaTes..." ?

**Comment by googol42** [comment]
I also had problems with the index not updating correctly. I recompiled/reinstalled albert again and now everything works. I also use the albert AUR package.

Maybe the AUR package is missing a dependency? I hope I am not mistaken, but couldn't it happen that a dependency changes which requires albert to be recompiled due to albert linking that dependency? And could a missing dependency in the AUR Package cause such problems, i.e. the package manager of your choice not realising that is has to reinstall albert?

**Comment by ManuelSchneid3r** [comment]
> Maybe the AUR package is missing a dependency? I hope I am not mistaken, but couldn't it happen that a dependency changes which requires albert to be recompiled due to albert linking that dependency? And could a missing dependency in the AUR Package cause such problems, i.e. the package manager of your choice not realising that is has to reinstall albert?

Not really. The indexing code does not require anything external. If pulled from aur you may have had an outdated version of albert.

**Comment by henrikgit** [comment]
Another unsuccessful test with: 

    $ pacman -Q albert 
    albert 0.21.0-1

What I tried this time:

1. Ran `$ sudo pacman -Rns albert`

1. Deleted `~/.config/albert` and `~/.config/albert.conf`.

1. Re-installed albert: `$ trizen -S albert`

1. After the installation ended, I restarted albert and configured it (but let the scan interval have the default value, i.e. 0). Then I let it run for 3-5 minutes or so.

1. I closed albert.

1. Restarted albert.

1. Created `/home/john/Downloads/22222222/123132`

1. Waited ca. 2 minutes.

1. Hit the albert hotkey and started entering "1231...".

The folder was not found.

**Comment by ManuelSchneid3r** [comment]
I finally took the effort to finally go through the files plugin. Indeed I found some bugs. Can somebody please check [albert-master](https://software.opensuse.org//download.html?project=home%3Amanuelschneid3r&package=albert-master)?

**Comment by henrikgit** [comment]
Would be happy to try and help, but I don't know the first thing about "checking albert-master" (which I suppose is the master repo?).

Is there a little guidance to start off?

**Comment by ManuelSchneid3r** [comment]
in the official repo is a package albert-master besides the usual albert which is build from git master.

They are built at obs: https://build.opensuse.org/project/show/home:manuelschneid3r
You can  add the repo and install the packages from here https://software.opensuse.org//download.html?project=home%3Amanuelschneid3r&package=albert-master

**Comment by ManuelSchneid3r** [comment]
sure, you dont need to code. just install albert and see if it's working 😄 . i guess this was just too much information. forget about all that and fire  `apt remove albert; apt install albert-master`, assuming that you are using a debian based system and added the repo before. if not, add the repo as described in the [obs software center]( https://software.opensuse.org//download.html?project=home%3Amanuelschneid3r&package=albert-master).

**Comment by henrikgit** [comment]
Got it, thanks. As indicated in the op, I am using Arch.

TL;DR: The issue and problem still exists.

What I did:
1. Went to https://software.opensuse.org//download.html?project=home%3Amanuelschneid3r&package=albert-master
2. Downloaded `albert-master-0.21.1-0-x86_64.pkg.tar`.
3. Per https://wiki.archlinux.org/title/Pacman#Additional_commands, I installed the pkg with `pacman -U ...`.
4. Restarted albert.
5. Waited like 3 minutes.
6. Created some new fake folders, e.g. `/home/john/Downloads/22222222/44444/` nor `/home/john/asdf/55555/`.
7. Neither get found after waiting roughly 2-3 minutes.

**Comment by ManuelSchneid3r** [comment]
Do you have any leftover plugins from earlier compilations? Is the plugin up to date?

**Comment by henrikgit** [comment]
> Do you have any leftover plugins from earlier compilations? 

Yes.

> Is the plugin up to date?

I would say yes.

![2023-08-11-1034 Files plugin is 0 22](https://github.com/albertlauncher/plugins/assets/6051516/b878a504-52b6-43c8-b144-995c5778c520)


Or should I check and list more details?

**Comment by ManuelSchneid3r** [comment]
Files version of latest master build should be 1.6

**Comment by ManuelSchneid3r** [comment]
check the paths of the used plugins listed on startup. it should use the one shipped with the prebuilt package. also i see there are a lot of plugins grayed out. please post the full output of a new instance started with `albert -d`

**Comment by henrikgit** [comment]
I also wondered about the grayed out plugins. I must admit I don't know what that is or how that happened.

Before I installed via `pacman -U`, I of course ran `pacman -R albert`. After I did that, there was package left in `/usr/lib/` with anything regarding albert, `$ ls /usr/lib/ | grep albert` produced no output.

I downloaded the package this way:

![2023-08-11-1106 albert-master downloaded version](https://github.com/albertlauncher/plugins/assets/6051516/c492acec-ac5b-4e32-8630-1364ef21860e)

If that was the correct way, please do say so.

Output of `albert -d`:

<details>

	$ albert -d 
			Albert version: 0.22.0
				Build date: Aug  7 2023 21:48:33
				Qt version: 6.5.2
				 Build ABI: x86_64-little_endian-lp64
	  Arch (build/current): x86_64/x86_64
	 Kernel (type/version): linux/6.4.9-arch1-1
						OS: Arch Linux
		 OS (type/version): arch/unknown
	 $QT_QPA_PLATFORMTHEME: qt5ct
			 Platform name: xcb
					  Font: Liberation Sans,11,-1,5,400,0,0,0,0,0,0,0,0,0,0,1,Regular
		   Binary location: /usr/bin/albert
					  $PWD: /home/john
					$SHELL: /bin/bash
					 $LANG: de_DE.UTF-8
				  Language: German
					Locale: de_DE
		 $XDG_SESSION_TYPE: x11
	  $XDG_CURRENT_DESKTOP: XFCE
		  $DESKTOP_SESSION: xfce
	  $XDG_SESSION_DESKTOP: xfce
				Icon theme: gnome
	11:07:47 [debg:albert] Creating crash indicator file
	11:07:47 [debg:albert] Checking for a running instance…
	11:07:47 [debg:albert] Creating local socket /home/john/.cache/albert/ipc_socket
	11:07:47 [debg:albert] Database: Connecting…
	11:07:47 [debg:albert] Database: Connected (     0 ms).
	11:07:47 [debg:albert] Database: Initializing…
	11:07:47 [debg:albert] Database: Initialized (     0 ms).
	11:07:47 [debg:albert] Database: Fetching activations…
	11:07:47 [debg:albert] Database: Activations fetched (     1 ms).
	11:07:47 [debg:albert]      1 ms fetching activations.
	11:07:47 [debg:albert]      0 ms computing usage scores.
	11:07:47 [debg:albert] Searching native plugins in /usr/lib/albert
	11:07:47 [debg:albert] Found valid native plugin /usr/lib/albert/libcalculator_muparser.so
	11:07:47 [debg:albert] Found valid native plugin /usr/lib/albert/liburlhandler.so
	11:07:47 [debg:albert] Found valid native plugin /usr/lib/albert/libwebsearch.so
	11:07:47 [debg:albert] Found valid native plugin /usr/lib/albert/libhash.so
	11:07:47 [debg:albert] Found valid native plugin /usr/lib/albert/libwidgetsboxmodel.so
	11:07:47 [debg:albert] Found valid native plugin /usr/lib/albert/libqmlboxmodel.so
	11:07:47 [debg:albert] Found valid native plugin /usr/lib/albert/libchromium.so
	11:07:47 [debg:albert] Found valid native plugin /usr/lib/albert/libclipboard.so
	11:07:47 [debg:albert] Found valid native plugin /usr/lib/albert/libcalculator_qalculate.so
	11:07:47 [debg:albert] Found valid native plugin /usr/lib/albert/libsnippets.so
	11:07:47 [debg:albert] Found valid native plugin /usr/lib/albert/libsystem.so
	11:07:47 [debg:albert] Found valid native plugin /usr/lib/albert/libfiles.so
	11:07:47 [debg:albert] Found valid native plugin /usr/lib/albert/libdatetime.so
	11:07:47 [debg:albert] Found valid native plugin /usr/lib/albert/libapplications_xdg.so
	11:07:47 [debg:albert] Found valid native plugin /usr/lib/albert/libssh.so
	11:07:47 [debg:albert] Found valid native plugin /usr/lib/albert/libdocs.so
	11:07:47 [debg:albert] Found valid native plugin /usr/lib/albert/libpython.so
	11:07:47 [debg:albert] Found valid native plugin /usr/lib/albert/libterminal.so
	11:07:47 [debg:albert] [     1 ms] QtPluginProvider::plugins
	11:07:47 [info:albert] Hotkey set to Ctrl+Space
	11:07:47 [debg:albert] Try loading the configured frontend 'widgetsboxmodel'.
	11:07:47 [debg:albert] [   136 ms] spent loading plugin 'widgetsboxmodel'
	11:07:47 [debg:albert] [     4 ms] spent loading plugin 'urlhandler'
	11:07:47 [debg:albert] [     2 ms] spent loading plugin 'websearch'
	11:07:47 [debg:albert] [     8 ms] spent loading plugin 'chromium'
	11:07:47 [debg:albert] Indexed bookmarks in      0 ms
	11:07:47 [info:albert] Indexing /home/john
	11:07:48 [debg:albert] [   499 ms] spent loading plugin 'files'
	11:07:48 [info:albert] 5 bookmarks indexed.
	11:07:48 [debg:albert] [     0 ms] spent loading plugin 'applications_xdg'

[From hereon forth it's just the `.desktop` files for the applications.]

</details>

**Comment by ManuelSchneid3r** [comment]
> I also wondered about the grayed out plugins.

the plugin is [not loaded](https://github.com/albertlauncher/albert/blob/ec1b86094337ebe4a6f7f90462347cf6d13d9613/src/settings/pluginwidget.cpp#L70)

> nothing left in `/usr/lib/` with anything regarding albert

plugins are looked up in ~/.local/, /usr/local/, /usr. in that order. if you ever fired a `make install` there may reside some old plugins as well.

> I downloaded the package this way:

id recommend to add the repository. otherwise you dont get updates.

**Comment by henrikgit** [comment]
I added the [repo for the Arch package](https://software.opensuse.org//download.html?project=home%3Amanuelschneid3r&package=albert-master#manualArch). Unfortunately, starting albert then does not work:

![2023-08-11-1438 albert failed QML component](https://github.com/albertlauncher/plugins/assets/6051516/e9636035-d251-4509-a62e-3861778c3102)

**Comment by henrikgit** [comment]
Still won't work, even with the Files plugin on v1.6.

![2023-08-11-1501 albert Files plugin version](https://github.com/albertlauncher/plugins/assets/6051516/609e1367-df90-42cd-9f39-ec2c1192511a)

Btw, what I found was, that while I click around in the plugins tab, all the listed plugin names are black.

But if I click somewhere else on my machine, meaning the settings window is out of focus, _only_ the activated & checkmarked plugin entries get grayed out. That happens regardless of the Frontend setting, so both times when it is set to "Widgets BoxModel" or "QML Box Model" (with a space?).

**Comment by ManuelSchneid3r** [comment]
I think i found the problem. OBS is currently building. Let me know if the recent commits fixed this issue.

**Comment by henrikgit** [comment]
No update found/performed yet when I run `pacman -Syu`. When is the new build supposed to be available?

**Comment by ManuelSchneid3r** [comment]
https://download.opensuse.org/repositories/home:/manuelschneid3r/Arch/x86_64/

According to the mdate the build yesterday has been shipped.

**Comment by henrikgit** [comment]
Thank you for the clarification. Sorry, I must have updated yesterday then. 

Anyway, still won't work. For the sake of clarity: https://youtu.be/o7Slv4-nmmQ

**Comment by henrikgit** [comment]
The video is supposed to show what I did for testing.

**Comment by ManuelSchneid3r** [comment]
I added some tests, all of them pass. also i tested on majaro and it works. in the video i see that your index is not updated. if the screenies above are still up to date, then your scan interval is set to 0 minutes. set it to at least 1 minute.

**Comment by henrikgit** [comment]
After I set the scan interval value to 1, it works.

Somewhat self-explanatory and tricky at the same time, as I had only changed the `Max Depth` when I configured my home path initially. Wasn't aware that changing the scan interval value from 0 to something higher than 0 is necessary.

Problem solved... thank you very much, Manuel. :)

**Comment by ManuelSchneid3r** [comment]
Great! Finally 😆 Thank you for the feedback
